### PR TITLE
Throw a nice error when we can't start the tar process 

### PR DESCRIPTION
**Pull request by b-barthel**
Throw a more actionable error in pipeline caching when we can't start Tar/7zip.

### CODE REVIEW

File: src/Test/L0/Plugin/TarUtilsL0.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using System.Threading.Tasks;
+using Xunit;
+using System.Threading;
+using Agent.Sdk;
+using Agent.Plugins.PipelineCache;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.PipelineCaching
+{
+    public  class TarUtilsL0 {
```
Comment (fadnavistanmay) [inline src/Test/L0/Plugin/TarUtilsL0.cs]: +1

**Comment by fadnavistanmay** [comment]
LGTM!

**Comment by fadnavistanmay** [comment]
@jtpetty / @damccorm - could you please look into this. Thanks.

File: src/Agent.Plugins/PipelineCache/TarUtils.cs [hunk 2]
```diff
@@ -127,7 +125,8 @@ public static class TarUtils
                 }
                 catch (Exception e)
                 {
-                    ExceptionDispatchInfo.Capture(e).Throw();
+                    // couldn't start the process, so throw a slightly nicer message about required dependencies:
+                    throw new InvalidOperationException($"Failed to start the required dependency '{process.StartInfo.FileName}'.  Please verify the correct version is installed and available on the path.", e);
```
Comment (jahsu-MSFT) [inline src/Agent.Plugins/PipelineCache/TarUtils.cs]: Maybe can also consider using StringUtil.Loc next time :)

Comment (b-barthel) [reply src/Agent.Plugins/PipelineCache/TarUtils.cs]: Yep larger work needed to localize pipeline caching.

**Pull request merged**